libs/content/workspace/src/tab/markdown_editor/editor.rs

```diff
@@ -397,3 +397,17 @@ impl Editor {
         None
     }
 }
+
+// used by task manager to offload serialization to a background thread
+impl Clone for Editor {
+    fn clone(&self) -> Self {
+        Self::new(
+            self.core.clone(),
+            &self.buffer.current.text,
+            self.file_id,
+            self.hmac,
+            self.needs_name,
+            self.appearance.plaintext_mode,
+        )
+    }
+}
```

Comment:
I'm not sure this is suitable behavior for `Clone`. I imagine this resets some portion of the state of the editor, and I would not expect clone to behave this way. I expect `Clone` to be a deep copy of all state unless there's some smart pointer business going on.

Reply:
Easy enough to implement a fn that describes whatever you're trying to do.